src/server/client.rs

```diff
@@ -165,7 +166,16 @@ async fn handle_job_cancel(
     let mut responses: Vec<(JobId, CancelJobResponse)> = Vec::new();
 
     let job_ids: Vec<JobId> = match selector {
-        JobSelector::All => state.jobs().map(|j| j.job_id).collect(),
+        JobSelector::All => state
+            .jobs()
+            .map(|job| job.make_job_info())
+            .filter(|job_info| {
+                vec![Status::Running, Status::Waiting]
```

Comment:
I would prefer a simpler check like

```rust
let status = job_status(&job_info);
status == Status::Running || status == Status::Waiting
```

Maybe the compiler could inline all this, but I think that making it simpler is also more readable :)

We could also introduce something like

```rust
impl Status {
    fn is_active(&self) -> bool {
         matches!(self, Self::Running | Self::Waiting)
    }
}
```

Reply:
I agree, creating vec over and over is too much. matches! is much better